index.ts

```diff
@@ -28,7 +42,17 @@ try {
   process.exit(1);
 }
 
-run(diff)
+let currentBranch = "";
+try {
+  currentBranch = execSync("git branch --show-current").toString();
+  currentBranch = currentBranch.replace(/[\r\n]/gm, '').trim();
+  currentBranch = currentBranch.substring(0, 8);
```

Comment:
How do we know it's the `substring 0,8`? It could be DEV-NNNNN at some point so this would break down or if we completed a card from a different board i.e. PROD-N?

Can we not instead just grab the board and number separately and then join later? Assume that the branch is always in the format: `board`-`issueNumber`-name-of-task

Reply:
It's a good point indeed. Since we've never worked on a different board, I assumed we're not going to start now. Also thought that since it took a year and a half to reach a 4-digit card number, that we still have years ahead of us before needing to worry about dev-10000.

The proposed change makes sense and I'll refactor it now.

Reply:
Understood :) think of the `substring` as hard-coding the first 8 characters whereas we want to dynamically extract the information to support different cases. We could also maybe think of other erroneous situations for example if there is no `board` / `issueNumber` extracted

Reply:
> Understood :) think of the `substring` as hard-coding the first 8 characters whereas we want to dynamically extract the information to support different cases. We could also maybe think of other erroneous situations for example if there is no `board` / `issueNumber` extracted

We don't know how will the name of the branch would look without a `board` or `issueNumber` and it is still going to be a string with dashes in-between so we can refactor when we have an example. For now, I'll only consider taking the `board` and `issueNumber` no matter how long they are. If you have any ideas  - let me know.

Reply:
For rare cases where there is no Jira board or/and Jira card number - we should use the last option - `[write own message]...`